### PR TITLE
Metadata kept in separate nuons instead of scripts

### PR DESCRIPTION
**Pull request by skelly37**
# Description

(description of your pull request here)

# Tests

Make sure you've done the following:

- [ ] Ensure your files have up to date metadata embedded inside
- [ ] (If submitting a module) Ensure all exported functions have documentation added to them
- [ ] (If submitting a module) Ensure all arguments to exported functions have documentation added to them
- [ ] (If submitting a module) Ensure all exported functions have examples added to them if usage is non obvious
- [ ] (If submitting a module) Ensure helper functions and variables are not exported from the module
- [ ] (If submitting a module) Ensure all exported functions have completion commands (unless non-feasible)
- [ ] (If submitting a standalone script) Ensure your script has a proper shebang at the top (`/usr/bin/env nu` should work in most circumstances)
- [ ] (If submitting a standalone script) Ensure your script has proper file permissions (Run `chmod +x` on the file before push)

### CODE REVIEW

File: generate-repo-cache.nu [hunk 1]
```diff
@@ -50,11 +46,18 @@ def add-optional-attributes [
...
+|where type == dir
+|get name
+|each {|it|
+    ls $it
+    |get name
+    |where $it =~ "*.nuon"
+    |each {|module|
+        check-required-attributes (add-optional-attributes (get-metadata $module))
+        |to nuon
+        |save repo-cache.nuon --append
```
Comment (skelly37) [inline generate-repo-cache.nu:61]: @Yethal any idea why does it get echo'ed instead of saving?

Comment (skelly37) [reply generate-repo-cache.nu]: I don't understand why the current solution fixed but it's fixed and I don't want to get into rabbit hole ig

File: modules/nupac/nupac.nu [hunk 2]
```diff
@@ -103,7 +85,7 @@ def packages-to-process [
 
 # downloads fresh repository cache
 def update-repo [] {
-    fetch https://raw.githubusercontent.com/skelly37/nupac/main/nupac.nuon
+    fetch https://raw.githubusercontent.com/skelly37/nupac/metadata-to-nuons/repo-cache.nuon
```
Comment (skelly37) [inline modules/nupac/nupac.nu:88]: revert to main before merge

Comment (Yethal) [reply modules/nupac/nupac.nu:88]: Maybe we could use an environment variable here and default it to main if it's not set

Comment (skelly37) [reply modules/nupac/nupac.nu:88]: And where would you declare/keep the variable?

**Pull request closed without merging**